tests/unit/dataquery/test_api.py

```diff
@@ -1,33 +1,222 @@
+
+        test = sorted(list(results_dict.keys()))
+        benchmark = sorted(list(self.results_dict.keys()))
+        self.assertTrue(test == benchmark)
+
+        # Add a ticker that does not have a "valid" time-series and will subsequently be
+        # removed. Confirm the series has been removed from the dictionary.
+        f_ticker = next(iter(results_dict.keys()))
+        shape = results_dict[f_ticker].shape
+        # Again, as described above, a series is not valid if all values are NoneType.
+        data = np.array([None] * (shape[0] * shape[1]))
+
+        results_dict['DB(JPMAQS,USD_FXXR_NSA'] = data.reshape(shape)
+        results_dict_USD = api.Interface.valid_ticker(self.results_dict,
+                                                      suppress_warning=True,
+                                                      debug=False)
+        # Ticker should be removed from the dictionary.
+        self.assertTrue('DB(JPMAQS,USD_FXXR_NSA' not in results_dict_USD.keys())
+
+    def test_dataframe_wrapper(self):
+
+        cids_dmca = ['AUD', 'CAD', 'CHF', 'EUR', 'GBP', 'JPY']
+        xcats = ['EQXR_NSA', 'FXXR_NSA']
+
+        tickers = [cid + '_' + xcat for xcat in xcats for cid in cids_dmca]
+
+        # After the time-series have been isolated for each ticker and all the tickers
+        # have been validated, aggregate the results, still held in a dictionary, into
+        # a single DataFrame where the columns will be the passed metrics, 'value' &
+        # 'grading' plus the standardised JPMaQS columns: 'cid', 'xcat', 'real_date'.
+
+        # The method api.Interface.valid_ticker() is not required given each ticker will
+        # be valid by design.
+        self.test_isolate_timeseries()
+
+        results_dict = self.results_dict
+        trial_df = api.Interface.dataframe_wrapper(_dict=results_dict, no_metrics=2,
+                                                   original_metrics=['value', 'grading'])
+
+        # Confirm the dictionary is a standardised DataFrame plus the respective metrics
+        # passed.
+        expected_columns = ['cid', 'xcat', 'real_date', 'value', 'grading']
+        self.assertTrue(sorted(list(trial_df.columns)) == sorted(expected_columns))
+
+        # Next confirm that tickers held in the DataFrame are the complete set: i.e all
+        # the tickers defined in the constructor. All tickers are valid. Therefore,
+        # confirm there is not any inadvertent leakage from constructing the DataFrame
+        # and all tickers are included.
+
+        tickers_df = list(trial_df['cid'] + '_' + trial_df['xcat'])
+        self.assertTrue(sorted(tickers_df) == sorted(tickers))
+
+        # Given the constructed nature of the DataFrame, confirm all values in the
+        # 'grading' column are equal to 1.0.
+        # Confirms the columns have the expected values.
+        grades_test = np.all(trial_df['grading'] == 1.0)
+        self.assertTrue(grades_test)
```

Comment:
Re-write to something like:
```python
self.assertIn(trial_df['grading'], [1.0])
```

Reply:
Missing implementation as well. Will show up in the test failure with the grading that isn't 1.0, hence the change.

Reply:
Just to be clear the grading will always be 1.0 as the grade is artificially created within the code but will make the change.